Gems/Atom/RPI/Code/Source/RPI.Public/Image/StreamingImageController.cpp

```diff
@@ -157,25 +205,191 @@ namespace AZ
             // So it's unnecessary to queue again.
             if (!queuedForMipExpand)
             {
-                m_mipExpandMutex.lock();
-                m_mipExpandQueue.emplace_back(context);
-                m_mipExpandMutex.unlock();
+                AZStd::lock_guard<AZStd::mutex> lock(m_mipExpandMutex);
+                m_mipExpandQueue.push(context);
             }
         }
 
-        void StreamingImageController::QueueExpandToMipChainLevel(StreamingImage* image, size_t mipChainIndex)
+        uint32_t StreamingImageController::GetExpandingImageCount() const
         {
-            image->QueueExpandToMipChainLevel(mipChainIndex);
+            return aznumeric_cast<uint32_t>(m_expandingImages.size());
         }
 
-        void StreamingImageController::TrimToMipChainLevel(StreamingImage* image, size_t mipChainIndex)
+        void StreamingImageController::SetMipBias(int16_t mipBias)
         {
-            image->TrimToMipChainLevel(mipChainIndex);
+            int16_t prevMipBias = m_mipBias;
+            m_mipBias = mipBias;
+            // if the mipBias increased (using smaller size of mips), go throw each streaming image and evict unneeded mips
+            if (m_mipBias > prevMipBias)
+            {
+                EvictUnusedMips();
+            }
+        }
+
+        int16_t StreamingImageController::GetMipBias() const
+        {
+            return m_mipBias;
         }
 
-        StreamingImageContextPtr StreamingImageController::CreateContextInternal()
+        StreamingImageContextPtr StreamingImageController::CreateContext()
         {
             return aznew StreamingImageContext();
         }
+
+        void StreamingImageController::ResetLowMemoryState()
+        {
+            m_lastLowMemory = 0;
+        }
+
+        bool StreamingImageController::EvictOneMipChain()
```

Comment:
Nit: Suggest `EvictOneMipLevel`. The "mip chain" is the set of all the mips of an image.

Reply:
The StreamingImage's expand or evict unit is one MipChain but not one mip level.